docs/advanced/testing.md

```diff
@@ -61,3 +61,76 @@ ResiliencePipelineDescriptor descriptor = pipeline.GetPipelineDescriptor();
 // ...
 ```
 <!-- endSnippet -->
+
+
+## Patterns and anti-patterns
+
+Over the years, many developers have used Polly in various ways. Some of these recurring patterns may not be ideal. This section highlights the recommended practices and those to avoid.
+
+### 1 - Recreating the policies inside unit tests
+
+Imagine that you have a class that receives a `ResiliencePipeline<T>` as a consturtor parameter for testability.
+
+❌ DON'T
+
+Recreate the strategey inside the unit test:
+
+<!-- snippet: testing-anti-pattern-1 -->
+```cs
+// Arrange
+var timeout = new ResiliencePipelineBuilder<HttpResponseMessage>()
+    .AddTimeout(TimeSpan.FromSeconds(1))
+    .Build();
+
+var mockDownstream = Substitute.For<IDownstream>();
+mockDownstream.CallDownstream(Arg.Any<CancellationToken>())
+.Returns((_) => { Thread.Sleep(5_000); return new HttpResponseMessage(); }, null);
+
+// Act
+var sut = new SUT(mockDownstream, timeout);
+await sut.RetrieveData();
+```
+<!-- endSnippet -->
+
+**Reasoning**:
+
+- Depending on the your strategy setup, the execution time might be greatly increased (from milliseconds to seconds).
+  - For example if you have a retry with 3 `MaxRetryAttempts` and 1 seconds `Delay`.
+- This violates one of the F.I.R.S.T testing principles, namely the **Fast** one.
+
+✅ DO
+
+Use `Empty` or mock `ExecuteAsync`:
+
+- If you want to test your to-be-decorated code like there is no strategy applied
+
+<!-- snippet: testing-pattern-1-1 -->
+```cs
+var timeoutMock = ResiliencePipeline<HttpResponseMessage>.Empty;
+```
+<!-- endSnippet -->
+
+- If you want to verify that your code handles success cases as expected
```

Comment:
```suggestion
- If you want to verify that your code handles success cases as expected:
```

Reply:
Not directly, if there is a desire to do it we might introduce some API into Polly.Testing package.

Something similar to:
https://github.com/App-vNext/Polly/blob/main/test/Polly.TestUtils/TestResilienceStrategy.cs

But I would rather wait for a feedback.